confluent_kafka/src/Consumer.c

```diff
@@ -825,6 +886,28 @@ static PyMethodDef Consumer_methods[] = {
       "  :raises: RuntimeError if called on a closed consumer\n"
 	  "\n"
 	},
+	{ "consume", (PyCFunction)Consumer_consume,
+	  METH_VARARGS|METH_KEYWORDS,
+	  ".. py:function:: consume([num_messages=None], [timeout=None])\n"
+	  "\n"
+	  "  Consume messages, calls callbacks and returns list of messages.\n"
+	  "\n"
+	  "  The application must check the returned :py:class:`Message` "
+	  "object's :py:func:`Message.error()` method to distinguish "
+	  "between proper messages (error() returns None), or an event or "
+	  "error for each :py:class:`Message` in the list (see error().code() "
+	  "for specifics).\n"
+	  "\n"
+	  "  .. note: Callbacks may be called from this method, "
+	  "such as ``on_assign``, ``on_revoke``, et.al.\n"
+	  "\n"
+	  "  :param int num_messages: Maximum number of messages to return.\n"
+	  "  :param float timeout: Maximum time to block waiting for message, event or callback.\n"
+	  "  :returns: A list of Message objects or None on timeout\n"
```

Comment:
Do we really need to use a different return type for timeout? An empty list seems easier to handle from the application's perspective.